sdk/identity/identity/samples/AzureCacheForRedis/ioredis.md

```diff
@@ -0,0 +1,273 @@
+- Familiarity with the [ioredis](https://github.com/luin/ioredis) and [Azure Identity for JavaScript](https://docs.microsoft.com/javascript/api/overview/azure/identity-readme?view=azure-node-latest) client libraries is assumed.
+
+#### Samples Guidance
+
+* [Authenticate with Azure AD - Hello World](#authenticate-with-azure-ad-hello-world):
+   This sample is recommended for users getting started to use Azure AD authentication with Azure Cache for Redis.
+* [Authenticate with Azure AD - Handle Reauthentication](#authenticate-with-azure-ad-handle-reauthentication):
+   This sample is recommended to users looking to build long-running applications and would like to handle reauthenticating with Azure AD upon token expiry.
+* [Authenticate with Azure AD - Using Token Cache](#authenticate-with-azure-ad-using-token-cache):
+  This sample is recommended to users looking to build long-running applications that would like to handle reauthenticating with a token cache. The token cache stores and proactively refreshes the Azure AD access token 2 minutes before expiry and ensures a non-expired token is available for use when the cache is accessed.
+
+#### Authenticate with Azure AD: Hello World
+
+This sample is intended to assist in authenticating a hosted Azure Cache for Redis instance with Azure AD via the ioredis client library. It focuses on displaying the logic required to fetch an Azure AD access token and to use it as the password when setting up the ioredis instance.
+
+##### Migration Guidance
+
+When migrating your existing application code to authenticate with Azure AD, replace the password input with Azure AD token. Azure Redis Cache name, username, Azure AD token, and use of SSL are required while connecting with the cache.
+The username will depend on whether you're using service principal, managed identity, or Microsoft username.
+
+Integrate the logic in your application code to fetch an Azure AD access token via the Azure Identity library, as shown below. Replace it with the password configuring/retrieving logic in your application code.
+
+```ts
+import Redis from "ioredis";
+import { DefaultAzureCredential } from "@azure/identity";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+async function main() {
+  // Construct a Token Credential from Identity library, e.g. ClientSecretCredential / ClientCertificateCredential / ManagedIdentityCredential, etc.
+  const credential = new DefaultAzureCredential();
+
+  // The scope will be changed for Azure AD Public Preview
+  const redisScope = "https://*.cacheinfra.windows.net:10225/appid/.default"
+  
+  // Fetch an Azure AD token to be used for authentication. This token will be used as the password.
+  let accessTokenObject = await credential.getToken(
+    redisScope
+  );
+
+  // Create ioredis client and connect to the Azure Cache for Redis over the TLS port using the access token as password.
+  const redis = new Redis({
+    username: process.env.REDIS_SERVICE_PRINCIPAL_NAME,
```

Comment:
The sample needs to provide clear instructions for where to get this value (I assume this means the name of the customer-created app registration that the role is bound to) and how to configure the role binding that the service requires, etc. Is this the SPN, or the name of the app registration.

Reply:
Configuration of the roles is mentioned in the pre-requisite section.

Reply:
Mentioned, but not explained and no clear instructions. What role binding do I apply to grant access to the redis instance?